packages/atomic/src/components/search/atomic-result-timespan/atomic-result-timespan.ts

```diff
@@ -0,0 +1,151 @@
+import {ValidatePropsController} from '@/src/components/common/validate-props-controller/validate-props-controller';
+import type {Bindings} from '@/src/components/search/atomic-search-interface/interfaces';
+import {createResultContextController} from '@/src/components/search/result-template-component-utils/context/result-context-controller';
+import {bindingGuard} from '@/src/decorators/binding-guard';
+import {bindings} from '@/src/decorators/bindings';
+import {errorGuard} from '@/src/decorators/error-guard';
+import type {InitializableComponent} from '@/src/decorators/types';
+import {InitializeBindingsMixin} from '@/src/mixins/bindings-mixin';
+import {LightDomMixin} from '@/src/mixins/light-dom';
+
+dayjs.extend(duration);
+
+/**
+ * The `atomic-result-timespan` component renders a target result number field value as a duration.
+ */
+@customElement('atomic-result-timespan')
+@bindings()
+export class AtomicResultTimespan
+  extends LightDomMixin(InitializeBindingsMixin(LitElement))
+  implements InitializableComponent<Bindings>
+{
+  /**
+   * The target result field.
+   * The component first looks for the field in the Result object, and then in the Result.raw object.
+   * It is important to include the necessary field in the `atomic-search-interface` component.
+   */
+  @property({reflect: true}) public field!: string;
+  /**
+   * The unit of measurement of the field value.
+   * Available units: https://day.js.org/docs/en/durations/creating
+   */
+  @property({reflect: true}) public unit = 'ms';
+  /**
+   * The format to apply to the result field value.
+   *
+   * By default, the format is HH:mm:ss when the duration is under a day, and it is an approximation when longer (days, months or years).
+   *
+   * The string displayed when there is an approximation can be modified with [localization](https://docs.coveo.com/en/atomic/latest/usage/atomic-localization/).
+   *
+   * Available formats: https://day.js.org/docs/en/durations/format
+   */
+  @property({reflect: true}) public format?: string;
+
+  @state() public bindings!: Bindings;
+  @state() public error!: Error;
+
+  private resultContext = createResultContextController(this);
+
+  constructor() {
+    super();
+
+    new ValidatePropsController(
+      this,
+      () => ({field: this.field, unit: this.unit, format: this.format}),
+      new Schema({
+        field: new StringValue({required: true, emptyAllowed: false}),
+        unit: new StringValue({
+          constrainTo: [
+            'milliseconds',
+            'ms',
+            'seconds',
+            's',
+            'minutes',
+            'm',
+            'hours',
+            'h',
+            'days',
+            'd',
+            'weeks',
+            'w',
+            'months',
+            'M',
+            'years',
+            'y',
+          ],
+        }),
+        format: new StringValue({required: false, emptyAllowed: false}),
+      })
```

Comment:
Do we need to recreate the schema at every component init?

Reply:
Indeed that's a bit brutal for a result template component that might be rendered multiple times in the page 😅